totalRP3/modules/register/characters/PlayerMapScan/PlayerMapPin.lua

```diff
@@ -0,0 +1,110 @@
+----------------------------------------------------------------------------------
+--- Total RP 3
+---    ---------------------------------------------------------------------------
+---    Copyright 2014 Sylvain Cossement (telkostrasz@telkostrasz.be)
+--- Copyright 2018 Renaud "Ellypse" Parize <ellypse@totalrp3.info> @EllypseCelwe
+---
+---    Licensed under the Apache License, Version 2.0 (the "License");
+---    you may not use this file except in compliance with the License.
+---    You may obtain a copy of the License at
+---
+---        http://www.apache.org/licenses/LICENSE-2.0
+---
+---    Unless required by applicable law or agreed to in writing, software
+---    distributed under the License is distributed on an "AS IS" BASIS,
+---    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+---    See the License for the specific language governing permissions and
+---    limitations under the License.
+----------------------------------------------------------------------------------
+
+---@type TRP3_API
+local _, TRP3_API = ...;
+local Ellyb = TRP3_API.Ellyb;
+---@type AddOn_TotalRP3
+local AddOn_TotalRP3 = AddOn_TotalRP3;
+
+--region Lua imports
+local huge = math.huge;
+--endregion
+
+--region Total RP 3 imports
+local loc = TRP3_API.loc;
+--endregion
+
+--region Ellyb imports
+local ORANGE = Ellyb.ColorManager.ORANGE;
+---endregion
+
+-- Create the pin template, above group members
+TRP3_PlayerMapPinMixin = AddOn_TotalRP3.MapPOIMixins.createPinTemplate(
+	AddOn_TotalRP3.MapPOIMixins.GroupedCoalescedMapPinMixin, -- Use coalesced grouped tooltips (show multiple player names)
+	AddOn_TotalRP3.MapPOIMixins.AnimatedPinMixin -- Use animated icons (bounce in)
+);
+
+-- Expose template name, so the scan can use it for the MapDataProvider
+TRP3_PlayerMapPinMixin.TEMPLATE_NAME = "TRP3_PlayerMapPinTemplate";
+
+--- This is called when the data provider acquire a pin, to transform poiInfo received from the scan
+--- into display info to be used to decorate the pin.
+function TRP3_PlayerMapPinMixin:GetDisplayDataFromPOIInfo(poiInfo)
+	local characterID = poiInfo.sender;
+	local displayData = {};
+
+	if TRP3_API.register.isUnitIDKnown(characterID) and TRP3_API.register.hasProfile(characterID) then
```

Comment:
Invert this check and bring the `else` block up here and do an early return inside of it, then 90% of the function won't need to be indented.